.gitlab-ci.yml

```diff
@@ -268,6 +268,28 @@ test:backend:integration:
     reports:
       junit: backend/results_integration_*.xml
 
+test:integration:
+  stage: test
+  needs:
+    - job: build:backend:docker
+      artifacts: false
+    - job: build:frontend:docker
+      artifacts: false
+  rules:
+    - if: $CI_COMMIT_REF_PROTECTED == "true"
```

Comment:
I think this rule correct: you don't need to have it from a protected branch in the caller pipeline (`mender-server` in this case).

Reply:
The condition is for the dependencies only. It requires a protected branch (or modifications to both frontend and backend subdirectories) to build all images.